hls4ml/backends/oneapi/passes/reshaping_templates.py

```diff
@@ -161,29 +161,64 @@ def format(self, node):
 
 # Transpose templates
 
-transpose_config_template = """struct config{index} : nnet::transpose_config {{
-    static const unsigned depth = {depth};
-    static const unsigned height = {height};
-    static const unsigned width = {width};
-    static constexpr unsigned perm[3] = {{{perm_str}}};
+transpose_config_template = """struct {config_name} : nnet::transpose_config {{
+    static constexpr unsigned dims = {dims};
+    static constexpr unsigned N = {N};
+    static constexpr std::array<unsigned, dims> from_shape = {{{from_shape}}};
+    static constexpr std::array<unsigned, dims> to_shape = {{{to_shape}}};
+    static constexpr std::array<unsigned, dims> perm = {{{perm}}};
+    static constexpr std::array<unsigned, dims> perm_strides = {{{perm_strides}}};
 }};\n"""
 
-transpose_function_template = 'nnet::transpose_{dim}<{input_t}, {output_t}, {config}>({input}, {output});'
-transpose_task_sequence_template = (
-    'task_sequence<nnet::transpose_{dim}_stream<{input_pipe}, {output_pipe}, {config}>> {name};'
-)
+transpose_function_template = 'nnet::transpose<{input_t}, {output_t}, {config}>({input}, {output});'
+transpose_task_sequence_template = 'task_sequence<nnet::transpose_stream<{input_pipe}, {output_pipe}, {config}>> {name};'
 transpose_include_list = ['nnet_utils/nnet_transpose.h', 'nnet_utils/nnet_transpose_stream.h']
 
 
+def permute_config_gen(name: str, shape: tuple[int, ...], perm: tuple[int, ...]):
```

Comment:
This is identical to the one introduced by the #1124, it should be placed in some utility module to be shared by all backends.

Reply:
I moved this to where conv instructions are created. Is this the best place? The actual code that is centralized actually becomes kind of small.

Reply:
Actually, I wonder if all the conv instruction functions, along with this one, should be moved to a new file, maybe `fpga_utils.py`? Functions like `generate_conv1d_line_buffer_fn` or `compute_conv2d_instructions` really are `@staticmethod` type, even if they are not labeled as such. They don't need to be in `FPGABackend`.